homeassistant/components/rest/binary_sensor.py

```diff
@@ -86,15 +102,35 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     if rest.data is None:
         raise PlatformNotReady
 
-    # No need to update the sensor now because it will determine its state
-    # based in the rest resource that has just been retrieved.
-    add_entities([RestBinarySensor(hass, rest, name, device_class, value_template)])
+    add_entities(
+        [
+            RestBinarySensor(
+                hass,
+                rest,
+                name,
+                device_class,
+                value_template,
+                force_update,
+                resource_template,
+            )
+        ],
+        True,
```

Comment:
This should not be `True`, now you're updating the data twice. That's why you had to update the count in the tests too...

```suggestion
        ]
```